File: server/Models/Report.cs
using System.ComponentModel.DataAnnotations;

namespace help.Models;

public class Report : RepoItem<int>
{
  public string Title { get; set; }
  [MaxLength(500)] public string Body { get; set; }
  [Range(1, 5)] public int Score { get; set; }
  [Url, MaxLength(1000)] public string ImgUrl { get; set; }
  public string CreatorId { get; set; }
  public int RestaurantId { get; set; }
  public Profile Reporter { get; set; }
}

File: client/src/models/Report.js
import { Profile } from "./Account.js";
import { DatabaseItem } from "./DatabaseItem.js";

export class Report extends DatabaseItem {
  constructor(data) {
    super(data)
    this.title = data.title
    this.body = data.body ?? ''
    this.score = data.score
    this.imgUrl = data.imgUrl ?? ''
    this.creatorId = data.creatorId
    this.restaurantId = data.restaurantId
    this.reporter = new Profile(data.reporter)
  }
}